.github/workflows/test.yml

```diff
@@ -10,29 +11,24 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest, ubuntu-24.04-arm]
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+
+    runs-on: ${{ matrix.os }}
+
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
-
-      - name: Install poetry
-        run: |
-          pip install poetry
-          poetry config virtualenvs.in-project true
-
-      - name: Set up cache for dependencies
-        uses: actions/cache@v4
```

Comment:
I don't want to give up cache, especially on MR pipelines, which should be as fast as possible. It seems [this can be set up](https://github.com/abatilo/actions-poetry?tab=readme-ov-file#workflow-example-cache-the-virtual-environment)